fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java

```diff
@@ -479,10 +476,10 @@ public Map<String, Object> update(final JsonCommand command) {
             this.accountNumber = StringUtils.defaultIfEmpty(newValue, null);
         }
 
-        if (command.isChangeInStringParameterNamed(ClientApiConstants.externalIdParamName, this.externalId)) {
+        if (command.isChangeInStringParameterNamed(ClientApiConstants.externalIdParamName, this.externalId.getValue())) {
             final String newValue = command.stringValueOfParameterNamed(ClientApiConstants.externalIdParamName);
             actualChanges.put(ClientApiConstants.externalIdParamName, newValue);
-            this.externalId = StringUtils.defaultIfEmpty(newValue, null);
+            this.externalId = new ExternalId(newValue);
```

Comment:
This is wrong! Please move this logic out of the entity. 

Also, creating a new external id please use the `ExternalIdFactory`.